Sources/StreamChatUI/ChatMessageList/ChatMessageListVC.swift

```diff
@@ -303,6 +303,31 @@ open class ChatMessageListVC:
         typingIndicatorView.isHidden = true
     }
 
+    /// Check if the current message being displayed should show the date separator.
+    /// - Parameters:
+    ///   - message: The message being displayed.
+    ///   - indexPath: The indexPath of the message.
+    /// - Returns: A Boolean value depending if it should show the date separator or not.
+    open func shouldShowDateSeparator(forMessage message: ChatMessage, at indexPath: IndexPath) -> Bool {
+        guard isDateSeparatorEnabled else {
+            return false
+        }
+        
+        let previousIndexPath = IndexPath(row: indexPath.row + 1, section: indexPath.section)
+        guard let previousMessage = dataSource?.chatMessageListVC(self, messageAt: previousIndexPath) else {
+            // If previous message doesn't exist show the separator as well.
+            return true
+        }
+        
+        // Only show the separator if the previous message has a different day.
+        let isDifferentDay = !Calendar.current.isDate(
+            message.createdAt,
+            equalTo: previousMessage.createdAt,
+            toGranularity: .day
+        )
+        return isDifferentDay
+    }
```

Comment:
The change looks nice 👌 Just thinking if it won't be the issue when switching to sectioned table view? 🤔

Reply:
Good point 🤔 I think it should not, this function should be able to be moved to the section side since it only depends on the current message and indexPath. I can also make it internal for the time being, WDYT?